packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -94,6 +97,30 @@ export default function createCortexHandler(): express.Router {
     proxyTo(`http://store-gateway.cortex.svc.cluster.local/store-gateway/ring`)
   );
   // cortex.use(`/alertmanager/ring`, proxyTo(`http://alertmanager.cortex.svc.cluster.local/multitenant_alertmanager/ring`)) // sharding currenctly disabled
+  //     `http://ruler.cortex.svc.cluster.local/api/v1/rules/${req.params.namespace}/${req.params.group}`,
+  //     req,
+  //     res
+  //   );
+  // });
+
+  // Being able to access this implies having access to cortex config for
+  // all tenants, i.e. this is a privileged / superuser action.
+  cortex.post("/runtime_config", setCortexRuntimeConfigHandler);
+
+  // This is the runtime config as set in the config map
+  cortex.get("/runtime_config_file", readCortexRuntimeConfigHandler);
```

Comment:
`runtime_config_file` suggests we get the contents of the runtime config file; however at least one difference is that we get a JSON document instead of a YAML document -- is that the only difference?

Reply:
I don't think that `runtime_config_file` necessarily implies that the client would get yaml. We could add a content type if a client wanted to get yaml, but I don't see the use case there. Because that file can be edited using kubectl (or numerous other ways), or if we've deployed a newer version of Cortex to an existing cluster that has a different limits schema, I want to catch the source of error early and provide that feedback in CI (during upgrades for example) and also to users. Without validating the read path too, we'd have no guarantee of the data shape on the client, while at the same time relying on a strict data shape and therefore we open the possibility to unknown or unexpected bugs.

Reply:
> may this validation ever remove or add keys transparently?
It could if we wanted it to